exchanges/okex/okex_wrapper.go

```diff
@@ -419,6 +443,136 @@ func (o *OKEX) FetchTicker(p currency.Pair, assetType asset.Item) (tickerData *t
 }
 
 // GetHistoricCandles returns candles between a time period for a set time interval
-func (o *OKEX) GetHistoricCandles(pair currency.Pair, a asset.Item, start, end time.Time, interval time.Duration) (kline.Item, error) {
-	return kline.Item{}, common.ErrFunctionNotSupported
+func (o *OKEX) GetHistoricCandles(pair currency.Pair, a asset.Item, start, end time.Time, interval kline.Interval) (kline.Item, error) {
+	if !o.KlineIntervalEnabled(interval) {
+		return kline.Item{}, kline.ErrorKline{
+			Interval: interval,
+		}
+	}
+
+	req := okgroup.GetSpotMarketDataRequest{
+		Start:        start.UTC().Format(time.RFC3339),
+		End:          end.UTC().Format(time.RFC3339),
+		Granularity:  o.FormatExchangeKlineInterval(interval),
+		InstrumentID: o.FormatExchangeCurrency(pair, a).String(),
+	}
+
+	candles, err := o.GetSpotMarketData(req)
```

Comment:
OKEX supports many asset types, this fails for any other asset type. Please support all asset types

Reply:
Added support for futures & swaps 
Margin uses the same endpoint as SPOT

Reply:
Also just realized i need to add in index support will get that done